src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SignatureContext.cs

```diff
@@ -43,9 +42,24 @@ public IEnumerable<ParameterSyntax> StubParameters
                     if (typeInfo.ManagedIndex != TypePositionInfo.UnsetIndex
                         && typeInfo.ManagedIndex != TypePositionInfo.ReturnIndex)
                     {
+                        SyntaxTokenList tokens = TokenList();
+
+                        // "out" parameters are implicitly scoped, so we can't put the "scoped" keyword on them.
+                        // All other cases of explicit parameters are only scoped when the "scoped" keyword is present.
+                        // When the "scoped" keyword is present, it must be present on all declarations.
+                        if (typeInfo.ScopedKind != ScopedKind.None && typeInfo.RefKind != RefKind.Out)
+                        {
+                            tokens = tokens.Add(Token(SyntaxKind.ScopedKeyword));
+                        }
+
+                        if (typeInfo.IsByRef)
+                        {
+                            tokens = tokens.Add(Token(typeInfo.RefKindSyntax));
```

Comment:
We didn't conditionally add this previously, was that a bug?

Reply:
Technically we were adding a token for `SyntaxKind.None`. That ends up being an empty string when we stringify everything, so it didn't do anything wrong, but it's not necessary (and we get marginally better formatting by not including it when not needed)

Reply:
Oh, we were just propagating details from the current declaration. This make sense.